diff --git a/tools/musl-gcc.specs.sh b/tools/musl-gcc.specs.sh
index 294e24f7..0677fb1c 100644
--- a/tools/musl-gcc.specs.sh
+++ b/tools/musl-gcc.specs.sh
@@ -16,6 +16,9 @@ cat <<EOF
 *libgcc:
 libgcc.a%s %:if-exists(libgcc_eh.a%s)
 
+*self_spec:
+%{!static:%{!shared:%{!r:%{!fno-PIE:%{!no-pie:-fPIE -pie}}}}} %{!r:%{!fno-PIC:%{!fno-pic:-fPIC}}}
+
 *startfile:
 %{!shared: $libdir/%{pie:S}crt1.o} $libdir/crti.o %{shared|pie:crtbeginS.o%s;:crtbegin.o%s}
 
